RecoLocalCalo/EcalRecProducers/plugins/DeclsForKernels.h

```diff
@@ -169,64 +171,40 @@ namespace ecal {
         MYMALLOC(activeBXs, size * bxvlength);
-        //cudaCheck(cudaMalloc((void**)&activeBXs, size * sizeof(BXVectorType)));
         MYMALLOC(acState, size);
-        //cudaCheck(cudaMalloc((void**)&acState, size * sizeof(char)));
 
         MYMALLOC(hasSwitchToGain6, size);
-        //cudaCheck(cudaMalloc((void**)&hasSwitchToGain6, size * sizeof(bool)));
         MYMALLOC(hasSwitchToGain1, size);
-        //cudaCheck(cudaMalloc((void**)&hasSwitchToGain1, size * sizeof(bool)));
         MYMALLOC(isSaturated, size);
-        //cudaCheck(cudaMalloc((void**)&isSaturated, size * sizeof(bool)));
 
         if (configParameters.shouldRunTimingComputation) {
           MYMALLOC(sample_values, size * svlength);
-          //cudaCheck(cudaMalloc((void**)&sample_values, size * sizeof(SampleVector)));
           MYMALLOC(sample_value_errors, size * svlength);
-          //cudaCheck(cudaMalloc((void**)&sample_value_errors, size * sizeof(SampleVector)));
           MYMALLOC(useless_sample_values, size * EcalDataFrame::MAXSAMPLES);
-          //cudaCheck(cudaMalloc((void**)&useless_sample_values, size * sizeof(bool) * EcalDataFrame::MAXSAMPLES));
           MYMALLOC(chi2sNullHypot, size);
-          //cudaCheck(cudaMalloc((void**)&chi2sNullHypot, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(sum0sNullHypot, size);
-          //cudaCheck(cudaMalloc((void**)&sum0sNullHypot, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(sumAAsNullHypot, size);
-          //cudaCheck(cudaMalloc((void**)&sumAAsNullHypot, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(pedestal_nums, size);
-          //cudaCheck(cudaMalloc((void**)&pedestal_nums, size * sizeof(char)));
 
           MYMALLOC(tMaxAlphaBetas, size);
-          //cudaCheck(cudaMalloc((void**)&tMaxAlphaBetas, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(tMaxErrorAlphaBetas, size);
-          //cudaCheck(cudaMalloc((void**)&tMaxErrorAlphaBetas, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(accTimeMax, size);
-          //cudaCheck(cudaMalloc((void**)&accTimeMax, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(accTimeWgt, size);
-          //cudaCheck(cudaMalloc((void**)&accTimeWgt, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(ampMaxAlphaBeta, size);
-          //cudaCheck(cudaMalloc((void**)&ampMaxAlphaBeta, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(ampMaxError, size);
-          //cudaCheck(cudaMalloc((void**)&ampMaxError, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(timeMax, size);
-          //cudaCheck(cudaMalloc((void**)&timeMax, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(timeError, size);
-          //cudaCheck(cudaMalloc((void**)&timeError, size * sizeof(SampleVector::Scalar)));
           MYMALLOC(tcState, size);
-          //cudaCheck(cudaMalloc((void**)&tcState, size * sizeof(TimeComputationState)));
         }
```

Comment:
```suggestion
        }
#undef MYMALLOc
```

Reply:
I can add that.
Actually I was thinking of replacing the `#define` with a lambda - would it make more sense ?

Reply:
I could come up with
```c++
        auto allocate = [cudaStream](auto& var, uint32_t size) {
          using element_type = typename std::remove_reference_t<decltype(var)>::element_type;
          var = cms::cuda::make_device_unique<element_type[]>(size, cudaStream);
        };
```

Suggestions for improvement ?

---

It would be simpler if we could write
```c++
        auto allocate = [cudaStream]<typename T>(T& var, uint32_t size) {
          var = cms::cuda::make_device_unique<T::element_type[]>(size, cudaStream);
        };
```
but that's C++20, so not for a while yet.

Reply:
Lambda would certainly be much better. I don't have ideas how to improve your suggestions though.